### PR TITLE
feat(tokens): add danger, text05, update text02

### LINKED ISSUE #3736: [Button] Support token & danger buttons fix
## Three tasks to make Danger button accessible again! 

- [x] Update `$support-01` token colors on dark themes. Final token values should be below. **This token is for error message text, 4.5:1 accessible.** 

| White | G10 | G90 | G100 |
| --- | --- | --- | --- |
| Red 60 | Red 60 | Red 50 | Red 50 |  

- [x] Add `$danger` token with below values. **This token is for danger button / icon / non-text usage, 3:1 accessible.** 

| White | G10 | G90 | G100 |
| --- | --- | --- | --- |
| Red 60 | Red 60 | Red 60 | Red 60 |  

- [ ] Fix Danger button's color to use `$danger` or Red 60 on all themes. It's currently using an out-of-palette color for accessibility sake, but now our palette's Red 60 should be. 

image

### LINKED ISSUE #3960: Add text05 tokens
With new color palette, there are now 3 tiers of 4.5:1 accessible text colors: 

| White & Gray 10 | Gray 90 & Gray 100 |
| --- | --- |
| Gray 100, Gray 80, Gray 60 | Gray 10, Gray 30, Gray 50 | 

Our current text tokens are below:

image

## To do:
- [x] Add `$text-05` token
We need to add a new text token for "tertiary text", but should it follow the numbering and become `$text-03` bumping Placeholder text to something else, or just be named `$text-05` to avoid changing usage of existing tokens? @aagonzales 
- [ ] Update website & design kit. To be tracked separately.

### PR DESCRIPTION
**Pull request by shixiedesign**
Closes https://github.com/carbon-design-system/carbon/issues/3960
Partially closes #3736

This PR adds `$danger` and `$text05` token, and updates the values of `$text02` token on light themes. 

#### Changelog

**New**

- `$danger`
- `$text-05` for tertiary text

**Changed**
- `$text-02` on light themes

#### Testing / Reviewing

Make sure **Text input** & **Button-danger** are seeing the updates. It should have the following color: 

Text input - White theme Copy

### CODE REVIEW

File: packages/themes/src/g10.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ import {
   gray10,
   gray20,
   gray30,
+  gray40
```
Comment (joshblack) [inline packages/themes/src/g10.js]:
```suggestion
  gray40,
```

Comment (joshblack) [inline packages/themes/src/g10.js]:
```suggestion
  gray40,
```

**Review by joshblack — CHANGES_REQUESTED** [review]

**Review by asudoh — APPROVED** [review]
LGTM 👍 from the perspective of the code - Thanks @shixiedesign!

**Review by emyarod — COMMENTED** [review]
are the new tokens going to be added to text input styles in this PR? for example I think the error icon is still using `$support-01` and it can be changed to `$danger` now right?

**Comment by aagonzales** [comment]
We should talk when you get in the office. These are working differently that I thought. All parts of the error state should be using `support-01`. Why are they being mixed here? I thought the new token was for the danger buttons, I think realize it affected error states as well. I'm not sure if it worth having to different color reds on one error state, can they all just used the same color?

image

**Comment by shixiedesign** [comment]
After talking to Anna, we agreed to keep button & icons to use `$danger` for better contrast, border and error text to use `$support-01`. I updated PR description accordingly. This is another review item for text input.

**Review by aagonzales — APPROVED** [review]
The tokens themselves are good to go! (got distracted by the image)

File: packages/themes/src/v9.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ export const interactive01 = '#3d70b2';
 export const interactive02 = '#5a6872';
 export const interactive03 = '#5a6872';
 export const interactive04 = '#3d70b2';
-
+export const danger = `#e0182d`;
```
Comment (joshblack) [inline packages/themes/src/v9.js]:
```suggestion
export const danger = '#e0182d';
```

**Review by joshblack — CHANGES_REQUESTED** [review]

**Review by joshblack — APPROVED** [review]

**Comment by shixiedesign** [comment]
@emyarod No this PR only adds the tokens! Sorry I created the confusion with the image of text input. I didn't touch the components so the tokens are not applied. The actual components work will be separate issues and PRs.

**Pull request merged**